geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseQueryPlan.scala

```diff
@@ -85,3 +88,44 @@ case class GetPlan(filter: HBaseFilterStrategyType,
     SelfClosingIterator(resultsToFeatures(get.get(ranges).iterator), get.close)
   }
 }
+
+case class MultiRowRangeFilterScanPlan(filter: HBaseFilterStrategyType,
+                                       table: TableName,
+                                       ranges: Seq[Scan],
+                                       remoteFilters: Seq[HBaseFilter] = Nil,
+                                       resultsToFeatures: Iterator[Result] => Iterator[SimpleFeature]) extends HBaseQueryPlan {
+
+  override def scan(ds: HBaseDataStore): CloseableIterator[SimpleFeature] = {
+    import scala.collection.JavaConversions._
+
+    val rowRanges = Lists.newArrayList[RowRange]()
+    ranges.foreach { r =>
+      rowRanges.add(new RowRange(r.getStartRow, true, r.getStopRow, false))
+    }
+    val sortedRowRanges = MultiRowRangeFilter.sortAndMerge(rowRanges)
+    val numRanges = sortedRowRanges.length
+    val numThreads = ds.config.queryThreads
+    // TODO: parameterize this?
+    val rangesPerThread = math.max(1,math.ceil(numRanges/numThreads*2).toInt)
+    // TODO: align partitions with region boundaries
+    val groupedRanges = Lists.partition(sortedRowRanges, rangesPerThread)
+
+    val groupedScans = groupedRanges.map { localRanges =>
+      val mrrf = new MultiRowRangeFilter(localRanges)
```

Comment:
https://cloud.google.com/bigtable/docs/hbase-differences#filters 
MultiRowRangeFilter is not available in Google Bigtable.  
@anthonyccri @jahhulbert-ccri this'll break things with Google Cloud Bigtable.